go/vt/mysqlctl/mysqld.go

```diff
@@ -1251,6 +1251,29 @@ func (mysqld *Mysqld) applyBinlogFile(binlogFile string, includeGTIDs mysql.GTID
 		args := []string{
 			"--defaults-extra-file=" + cnf,
 		}
+
+		// We disable super_read_only, in case it is in the default MySQL startup
+		// parameters.  We do it blindly, since this will fail on MariaDB, which doesn't
+		// have super_read_only This is safe, since we're restarting MySQL after the restore anyway
+		log.Infof("applyBinlogFile: disabling super_read_only")
+		resetFunc, err := mysqld.SetSuperReadOnly(false)
+		if err != nil {
+			if strings.Contains(err.Error(), mysql.ERUnknownSystemVariable.ToString()) {
+				log.Warningf("Restore: server does not know about super_read_only, continuing anyway...")
+			} else {
+				log.Errorf("Restore: unexpected error while trying to set super_read_only: %v", err)
```

Comment:
This is not the `Restore` func, so the log message needs to be adjusted.

Reply:
done